authentik/lib/utils/email.py

```diff
@@ -0,0 +1,40 @@
+"""Email utility functions"""
+
+
+def mask_email(email: str | None) -> str | None:
+    """Mask email address for privacy
+
+    Args:
+        email: Email address to mask
+    Returns:
+        Masked email address or None if input is None
+    Example:
+        mask_email("myname@company.org")
+        'm*****@c******.org'
+    """
+    if not email:
+        return None
+
+    local, domain = email.split("@")
```

Comment:
Should add error handling if this function ever receives email addresses like `foo` or `foo@bar` since both of those are technically valid